### PR TITLE
User search

### PR DESCRIPTION
**Pull request by skylerwharton**
r @aharelick 
- Users can search for other users by first or last name.
- Added `@property` decorator to two `User` methods.

Currently, search bar is on a dummy page (`search.html`) at a dummy route (`/search`); in the future, it will be in the nabber.

screen shot 2015-11-04 at 9 34 13 pm

### CODE REVIEW

File: app/main/views.py [hunk 1]
```diff
@@ -1,7 +1,33 @@
...
+
+
+@main.route('/search')
+def search():
+    return render_template('main/search.html')
+
+
+@main.route('/search/<query>')
+def search_query(query):
+    looking_for = "%" + query + "%"
```
Comment (aharelick) [inline app/main/views.py]: single spaces when possible

File: app/main/views.py [hunk 2]
```diff
@@ -1,7 +1,33 @@
...
+    users = User.query.filter((User.first_name.ilike(looking_for)) |
+                              User.last_name.ilike(looking_for))\
+        .order_by(User.first_name).all()
+    data = dict()
+    data["results"] = []
```
Comment (aharelick) [inline app/main/views.py]: let's use some sexy list comprehension here and say something like

``` python
data['results'] = [{ 'title': u.full_name, 'url': '/account/profile/' + u.id } for u in users]
```

(may be a string/int concat problem in there)

**Comment by aharelick** [comment]
If you could go through and replace all the double quotes with single quotes in the python that would be great. Just a small list comprehension comment and then I think you're good to go.

File: app/main/views.py (hunk 1 above)
Comment (skylerwharton) [inline app/main/views.py]: Done

File: app/main/views.py (hunk 2 above)
Comment (skylerwharton) [inline app/main/views.py]: Done

**Comment by skylerwharton** [comment]
@aharelick ok, I think I'm done with everything.

**Comment by aharelick** [comment]
send it to the cloud :shipit:

**Comment by aharelick** [comment]
@weronica, adding the `@property` decorator to `full_name` and `is_admin` in https://github.com/hack4impact/women-veterans-rock/commit/574fe0645dbc7db48df6c0a66fe338f27f6c798a actually broke every reference to those two methods. I like the idea, but there's a ton of references to `user.full_name()`, so I think we should remove all of them and maybe make a separate pull request later.

**Pull request merged**